bochs/cpu/cpu.cc

```diff
@@ -338,9 +415,7 @@ void BX_CPP_AttrRegparmN(2) BX_CPU_C::repeat(bxInstruction_c *i, BxRepIterationP
       }
       if (RCX == 0) return;
 
-#if BX_DEBUGGER == 0
-      if (BX_CPU_THIS_PTR async_event)
-#endif
+      if (BX_CPU_THIS_PTR async_event || bx_dbg.debugger_active)
```

Comment:
we basically not allow traces when bx_dbg.debugger_active is set because debugger doesn't work well with them (many things are checked in main loop and not in the instruction handlers).
probably the easiest way to stop trace everywhere will be use smth like STOP_TRACE bit in async_event
May be invent one called ASYNC_EVENT_IN_DEBUGGER, set it from cpu_loop_debugger and will automatically cause exist to main loop after every instruction due to async_event being set

Reply:
done